packages/bits/src/lib/wizard-v2/wizard-step-header/wizard-step-header.component.ts

```diff
@@ -99,70 +101,62 @@ export class WizardStepHeaderComponent extends CdkStepHeader implements AfterVie
     }
 
     /** Focuses the step header. */
-    focus(): void {
+    public focus(): void {
         this._focusMonitor.focusVia(this._elementRef, "program");
     }
 
     /** Returns string label of given step if it is a text label. */
-    _stringLabel(): string | null {
+    public get stringLabel(): string | null {
         return this.label instanceof WizardStepLabelDirective ? null : this.label;
     }
 
     /** Returns WizardStepLabel if the label of given step is a template label. */
-    _templateLabel(): WizardStepLabelDirective | null {
+    public get templateLabel(): WizardStepLabelDirective | null {
         return this.label instanceof WizardStepLabelDirective ? this.label : null;
     }
 
-    _getDefaultTextForState(state: StepState): string {
-        let returnedState: StepState = state;
-        switch (state) {
-            case "number":
-                returnedState = `${this.index + 1}`;
-                break;
-
-            case "edit":
-                returnedState = "create";
-                break;
+    public get currentStepStateIconConfig(): IWizardCurrentStepStatusIconConfig | undefined {
```

Comment:
It would be great to have a state to iconConfig map that gets updated whenever the config changes. That way the all these if checks could be replaced with something like this:

```
return this.stateIconConfigMap[this.state];
```

Reply:
Do you mean something like this? This also works.
![image](https://user-images.githubusercontent.com/22597043/121352596-b316d600-c935-11eb-9160-103e2c5f882f.png)

What is the benefit of setting up a separate map over conditionally using the existing wizard config?

Reply:
For instance, if angular will ever change the name of the step status from `edit` to something else, this will bread (as well as the code snippet with conditionals though). The thing is I can utilize the STEP_STATE enum, which I actually forgot about. It is provided by angular and I can use it in my conditionals, while I don't think it is possible to make the enum values be the object keys. We could construct the map object from the enum though, but do you think this is worth the effort?